test/index.spec.js

```diff
@@ -0,0 +1,474 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const isNode = require('detect-node')
+const parallel = require('async/parallel')
+const series = require('async/series')
+
+const { Key } = require('interface-datastore')
+const { Record } = require('libp2p-record')
+
+const DatastorePubsub = require('../src')
+const { connect, waitFor, waitForPeerToSubscribe, spawnDaemon, stopDaemon } = require('./utils')
+
+// Always returning the expected values
+// Valid record and select the new one
+const smoothValidator = {
+  validate: (data, peerId, callback) => {
+    callback(null, true)
+  },
+  select: (receivedRecod, currentRecord, callback) => {
+    callback(null, 0)
+  }
+}
+
+describe('datastore-pubsub', function () {
+  this.timeout(60 * 1000)
+
+  if (!isNode) return
+
+  let ipfsdA = null
+  let ipfsdB = null
+  let ipfsdAId = null
+  let ipfsdBId = null
+
+  // spawn daemon
+  beforeEach(function (done) {
```

Comment:
There doesn't appear to be anything custom going on for the tests in regards to the daemons, can we just spawn them once in `before` instead of each test and then just stop them at the end? This is expensive.

Reply:
I am running each test with a new daemon to guarantee that the datastore does not have any data, or it would not be a unitary test. 

Another option that I thought of now. I can have a counter which is incremented on `afterEach` and the key I use to create the records would have the counter appended. What do you think? This way, I could use the same daemon for all the tests.

Reply:
Yes, I think using a unique key for each of the tests would be a more performant way to isolate the tests. If there are concerns about data collision I would test that separately and avoid creating new systems for each test. 

It might be good to remove `test` from `pre-push` as well with how long the tests currently take and just let ci handle running tests to avoid potential frustration for future contributors.

Reply:
Agreed, I will change it that way 🙂